compiler/noirc_evaluator/src/ssa/acir_gen/mod.rs

```diff
@@ -431,6 +431,9 @@ impl<'a> Context<'a> {
         }
 
         warnings.extend(return_warnings);
+
+        // Add the warnings from the alter Ssa passes
+        warnings.extend_from_slice(&ssa.warnings);
```

Comment:
Not a big fan of storing warnings within `Ssa` since it means we have to later remember to add them here. It looks like we'll also need to add this line in `convert_brillig_main` in case we have more warnings in the future which can occur in brillig as well.

Reply:
Moved out, but now I have to propagate them in return values